src/coffeescript.coffee

```diff
@@ -181,10 +181,8 @@ exports.tokens = withPrettyErrors (code, options) ->
 # return the AST. You can then compile it by calling `.compile()` on the root,
 # or traverse it by using `.traverseChildren()` with a callback.
 exports.nodes = withPrettyErrors (source, options) ->
-  if typeof source is 'string'
-    parser.parse lexer.tokenize source, options
-  else
-    parser.parse source
+  source = lexer.tokenize source, options if typeof source is 'string'
+  parser.parse(source).body
```

Comment:
As described, `CoffeeScript.nodes()` return `.body` to preserve backwards compatibility